cmd/di.go

```diff
@@ -509,3 +496,37 @@ func (di *Dependencies) bootstrapNATComponents(options node.Options) {
 		di.NATPinger = &traversal.NoopPinger{}
 	}
 }
+
+// FallbackResolverFactory creates a fallback resolver given the params
+func FallbackResolverFactory(ipResolver ip.Resolver, options node.OptionsLocation, configDirectory string) (location.Resolver, error) {
```

Comment:
Perhaps we can start working on packaging things by features. I think this func a fairly good candidate to be placed in `location`. Perhaps something like `location/di.go | container.go | config.go` for the configuration that merely assembles things for the app context.